Makefile

```diff
@@ -210,7 +210,7 @@ test-local:
 
 .PHONY: test-e2e
 test-e2e: ## Runs all Thanos e2e docker-based e2e tests from test/e2e. Required access to docker daemon.
-test-e2e: docker
```

Comment:
I think it makes sense. I also now remember why `make docker`  does not work for you - you build for Mac and copy binary to linux based image on docker, so that's why.

Let's have this but also ,can you comment on `make docker` that it work on Linux only, -or even better: Fail if you are on different ARCH/ OS? ((:

Reply:
Yes, haha. That makes sense.

**Proposal**: Another option I was considering, adjusting the `make docker` target to default to `make docker-multi-stage` on non linux x86_64 machine. Something along the lines of:

```make
.PHONY: docker
docker: ## Builds 'thanos' docker with no tag.
ifeq ($(OS)_$(ARCH), linux_x86_64)
docker: build
	@echo ">> copying Thanos from $(PREFIX) to ./thanos_tmp_for_docker"
	@cp $(PREFIX)/thanos ./thanos_tmp_for_docker
	@echo ">> building docker image 'thanos'"
	@docker build -t "thanos" .
	@rm ./thanos_tmp_for_docker
else
docker: docker-multi-stage
endif
```